src/terms/nonlocal.jl

```diff
@@ -193,47 +195,66 @@ function build_projection_vectors(basis::PlaneWaveBasis{T}, kpt::Kpoint,
 end
 
 """
-Build form factors (Fourier transforms of projectors) for an atom centered at 0.
+Build Fourier transform factors for all orbitals of an atom centered at 0.
 """
-function build_form_factors(psp, G_plus_k::AbstractVector{Vec3{TT}}) where {TT}
+function build_form_factors(psp, psp_fun::Function,                     
+                            count_n_fun_radial::Function, 
```

Comment:
Don't take the count functions or the psp_fun as input, they are implied by psp (just as it was before). Type psp as NormConservingPsp